packages/styled-base/types/index.d.ts

```diff
@@ -53,41 +53,68 @@ export interface StyledComponent<InnerProps, StyleProps, Theme extends object>
 }
 
 type ReactClassPropKeys = keyof React.ClassAttributes<any>
-export interface CreateStyledComponentBase<
+export type CreateStyledComponentBase<
   InnerProps,
   ExtraProps,
   StyledInstanceTheme extends object
-> {
-  <
-    StyleProps extends Omit<
-      Overwrapped<InnerProps, StyleProps>,
-      ReactClassPropKeys
-    > = Omit<InnerProps & ExtraProps, ReactClassPropKeys>,
-    Theme extends object = StyledInstanceTheme
-  >(
-    ...styles: Array<Interpolation<WithTheme<StyleProps, Theme>>>
-  ): StyledComponent<InnerProps, StyleProps, Theme>
-  <
-    StyleProps extends Omit<
-      Overwrapped<InnerProps, StyleProps>,
-      ReactClassPropKeys
-    > = Omit<InnerProps & ExtraProps, ReactClassPropKeys>,
-    Theme extends object = StyledInstanceTheme
-  >(
-    template: TemplateStringsArray,
-    ...styles: Array<Interpolation<WithTheme<StyleProps, Theme>>>
-  ): StyledComponent<InnerProps, StyleProps, Theme>
-}
-export interface CreateStyledComponentIntrinsic<
+> =
+  // this "reversed" condition checks if StyledInstanceTheme was already parametrized when using CreateStyled
+  object extends StyledInstanceTheme
+    ? {
```

Comment:
I think both cases here could be unified (the only difference is that 1 version should allow for second type parameter and the other one should have it already baked in), but I'm having slightly hard time to figure out how to do this properly right now

Reply:
One should allow, and one should not allow, so I think it's impossible to unify those cases.